packages/syntax-core/src/IconButton/IconButton.tsx

```diff
@@ -73,13 +76,15 @@ const IconButton = forwardRef<HTMLButtonElement, IconButtonProps>(
     }: IconButtonProps,
     ref,
   ) => {
+    const isMounted = useIsMounted();
+
     return (
       <button
         aria-label={accessibilityLabel}
         data-testid={dataTestId}
         type="button"
         title={tooltip}
-        disabled={disabled}
+        disabled={!isMounted.current || disabled}
```

Comment:
What do you think the implications of this change will be re: SEO robots?  It seems like the impact from this change specifically would be small, regardless.  That said, if this behavior needs to expand to other Buttons, would returning all of our buttons with `disabled` in the html response from the server affect the way an SEO robot sees/scores our pages?

It _seems_ like it won't be a big issue because these are buttons, and not links with `href` tags, so their behavior is dependent on js runtime behavior.  I just want to raise the concept for us to ponder a moment before going forward with this approach.

Reply:
@christianvuerings follow up and use Google pagerank tools - SEO

Reply:
@benjroy seems like crawlers like Google bot do not click buttons on the page, only links:
* https://support.google.com/webmasters/thread/40990043/does-google-bot-click-buttons?hl=en#:~:text=They%20don't%20click%20but,they%20won't%20crawl%20it.
* https://obrienmedia.co.uk/seo/googlebot-does-not-click-on-buttons-how-does-this-affect-seo

I am keeping any `ButtonLink` changes out of this PR but it seem like a possible solution would be to use `pointer-events: none` on the link. That way GoogleBot will still craw the link but the user can't click it until hydration completed
* https://github.com/microsoft/playwright/issues/12298

Reply:
> I am keeping any ButtonLink changes out of this PR but it seem like a possible solution would be to use pointer-events: none on the link.

Should we make that behavior consistent across the different components that need to look at this?  I.e. if we put something in that does `pointer-events: none`, could that same behavior be applied and work for all the components in this PR that are currently getting changes to the their `disabled` prop?

If so, then maybe we should go with that direction.  It seems like the solution could be applied more generically if that were the case?